sc/src/main/javascript/bundles/experiment-page/src/experiment-design/ExperimentDesignTable.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import ReactTable from 'react-table'
+import {uniq, curry} from 'lodash'
+import toPlural from 'pluralize'
+import '!style-loader!css-loader!react-table/react-table.css'
+import '!style-loader!css-loader!./experiment-table.css'
+
+const aggregateText = (name, vals) => {
+  const xs = uniq(vals)
+  return (
+    xs.length === 1 || (xs.length < 5 && xs.join(', ').length < 30)
+      ? xs.join(', ')
+      : toPlural(name.toLowerCase() , xs.length, true)
+  )
+}
+
+const ExperimentDesign = ({
+                            data,
+                            headers,
+                            options={}
+                          }) => (
+  <ReactTable
+    columns={
+      headers.map((headerGroup,ix)=> (
+        {
+          Header: headerGroup.name,
+          columns:
+            headerGroup.values.map((header, jx) => ({
+              aggregate: curry(aggregateText, 2)(header),
```

Comment:
I don't really understand what this aggregate function is for, so I haven't really touched it...

Reply:
It’s a function that, given a name and an array of values, if the array is 1 or less than 30 characters long when you join them with commas, it displays the full list. Otherwise it pluralizes it:
```
> aggregateText(`tissue`, [`lung`])
'lung'

> aggregateText(`tissue`, [`lung`, `heart`, `liver`])
'lung, heart, liver'

> aggregateText(`tissue`, [`lung`, `heart`, `liver`, `pancreas`, `bladder`, `skin`, `bone marrow`])
'7 tissues'
```

Reply:
The function is part of `ReactTable` API: [see Pivoting and Aggregation in the docs](https://react-table.js.org/#/story/readme).